analysis/runtime/src/runtime/scoped_runtime.rs

```diff
@@ -147,12 +168,14 @@ impl ExistingRuntime for BackgroundThreadRuntime {
     fn send_event(&self, event: Event) {
         match self.finalized.get() {
             None => {
-                // `.unwrap()` as we're in no place to handle an error here,
-                // unless we should silently drop the [`Event`] instead.
-                self.tx.send(event).unwrap();
+                // # Async-signal-safety: `push_event` is safe if `can_sleep == false`
+                self.push_event(event, false);
             }
             Some(()) => {
-                // Silently drop the [`Event`] as the [`BackgroundThreadRuntime`] has already been [`BackgroundThreadRuntime::finalize`]d.
+                // Silently drop the [`Event`] as the [`BackgroundThreadRuntime`]
+                // has already been [`BackgroundThreadRuntime::finalize`]d.
+                //
+                // # Async-signal-safety: Should be fine even after the end of `main`.
```

Comment:
```suggestion
                // Silently drop the [`Event`] as the [`BackgroundThreadRuntime`]
                // has already been [`BackgroundThreadRuntime::finalize`]d.
                //
                // # Async-signal-safety: `skip_event(_, AfterMain)` is NOT SAFE;
                // however, see the comment in `skip_event` for an explanation
                // of why this will probably be okay in practice.
```